chainio/clients/elcontracts/reader.go

```diff
@@ -230,6 +279,10 @@ func (r *ELChainReader) CalculateDelegationApprovalDigestHash(
 func (r *ELChainReader) CalculateOperatorAVSRegistrationDigestHash(
 	opts *bind.CallOpts, operator gethcommon.Address, avs gethcommon.Address, salt [32]byte, expiry *big.Int,
 ) ([32]byte, error) {
+	if r.avsDirectory == nil {
+		return [32]byte{}, errors.New("AVSDirectory contract not provided")
```

Comment:
let's start implementing custom errors?
Here would be something like
```
type NoContractError struct {
   Contract string
}
```
or maybe even we can have an enum instead of string to make it explicit

Reply:
yep. sounds good. we can change in separate PR.